consensus/pacemaker/pacemaker.go

```diff
@@ -0,0 +1,280 @@
+package pacemaker
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	consensusTelemetry "github.com/pokt-network/pocket/consensus/telemetry"
+	typesCons "github.com/pokt-network/pocket/consensus/types"
+	"github.com/pokt-network/pocket/runtime/configs"
+	"github.com/pokt-network/pocket/shared/codec"
+	"github.com/pokt-network/pocket/shared/modules"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+const (
+	DefaultLogPrefix    = "NODE"
+	pacemakerModuleName = "pacemaker"
+
+	// A buffer around the pacemaker timeout to avoid race condition; 30ms was arbitrarily chosen
+	timeoutBuffer = 30 * time.Millisecond
+
+	NewRound = typesCons.HotstuffStep_HOTSTUFF_STEP_NEWROUND
+	Propose  = typesCons.HotstuffMessageType_HOTSTUFF_MESSAGE_PROPOSE
+)
+
+type Pacemaker interface {
+	modules.Module
+	PacemakerDebug
+
+	ShouldHandleMessage(message *typesCons.HotstuffMessage) (bool, error)
+
+	RestartTimer()
+	NewHeight()
+	InterruptRound(reason string)
+}
+
+var (
+	_ modules.Module = &paceMaker{}
+	_ PacemakerDebug = &paceMaker{}
+)
+
+type paceMaker struct {
```

Comment:
We're mixing the use of `paceMaker` and `pacemaker` throughout. Let's go with the latter.